### PR TITLE
Julian additions

### PR DESCRIPTION
**Pull request by jl-n**
Added "powered by" message and share panel with display logic

### CODE REVIEW

File: src/components/SharePanel/SharePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      if (w <= 1600) {
+        return 'small' as Size
+      } else if (w <= 2000) {
+        return 'medium' as Size
+      } else {
+        return 'large' as Size
+      }
+    })(window.innerWidth)
+
+    const shareTitle = 'I just learned how to develop apps with Apollo and GraphQL'
```
Comment (marktani) [inline src/components/SharePanel/SharePanel.tsx]: given that this panel is used during the exercises, we could use another share title. Something like "Currently learning Apollo and GraphQL"?

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -208,19 +210,22 @@ class App extends React.Component<Props, State> {
           {this.state.storedState.user && this.state.storedState.user.projectId &&
           <div
             className={`
-              fixed bottom-0 left-0 flex fw3 items-center justify-center bg-accent pointer ${styles.serverButton}
+              fixed bottom-0 left-0 flex fw3 items-center justify-center flex-column bg-accent pointer ${styles.serverButton}
```
Comment (marktani) [inline src/components/App/App.tsx]: is this needed?

File: src/components/SharePanel/SharePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const shareUrl = encodeURIComponent(window.location.origin)
+
+    return (
+      <div className='flex justify-center'>
+        {this.state.display &&
+        <div className={'right-0 mt5 ph3 bg-black-05-opaque ' + (displayMode !== 'small' ? 'fixed' : '')}
+             style={{top: '18vh', maxWidth: (displayMode === 'medium' ? '16rem' : '40rem')}}>
+
+          <h3 className='accent' style={{fontWeight: 400}}>Liked Learn Apollo So far?</h3>
```
Comment (marktani) [inline src/components/SharePanel/SharePanel.tsx]: `so far`

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -51,8 +52,9 @@ class App extends React.Component<Props, State> {
   }
 
   componentDidMount() {
-    window.addEventListener('scroll', this.onScroll, false)
+    if(getStoredState().initialLoadTimestamp == null) update(['initialLoadTimestamp'], Date.now());
```
Comment (marktani) [inline src/components/App/App.tsx]: please wrap the if body with `{}`

File: src/components/SharePanel/SharePanel.tsx (hunk 1 above)
Comment (jl-n) [reply src/components/SharePanel/SharePanel.tsx]: I agree

File: src/components/SharePanel/SharePanel.tsx (hunk 3 above)
Comment (jl-n) [reply src/components/SharePanel/SharePanel.tsx]: Agreed

File: src/components/App/App.tsx (hunk 4 above)
Comment (jl-n) [reply src/components/App/App.tsx]: Oops, have done so now

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -232,7 +240,9 @@ class App extends React.Component<Props, State> {
           {this.props.children}
           {previousSubchapter &&
           <div
-            className={`${styles.jump} ${styles.jumpLeft} ${this.state.expandNavButtons ? styles.jumpActive : ''} ${this.state.showLayover ? styles.layoverPadding : ''} z-0`}
+            className={`${styles.jump} ${styles.jumpLeft} ${this.state.expandNavButtons
```
Comment (schickling) [inline src/components/App/App.tsx:243]: use the classnames package here

File: src/components/SharePanel/SharePanel.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    })(window.innerWidth)
+
+    const shareTitle = 'Learning how to develop apps with Apollo and GraphQL using LearnApollo!'
+    const shareUrl = encodeURIComponent(window.location.origin)
+
+    return (
+      <div className='flex justify-center'>
+        <div
+          className={'right-0 mt5 ph3 bg-black-05-opaque ' + (displayMode !== 'small' ? 'fixed' : '')}
```
Comment (schickling) [inline src/components/SharePanel/SharePanel.tsx:52]: possibly use classnames

File: src/components/SharePanel/SharingPanel.module.styl [hunk 7]
```diff
@@ -0,0 +1,41 @@
+.button
+  padding 15px
+  font-size 1rem
+  // font-weight 300
```
Comment (schickling) [inline src/components/SharePanel/SharingPanel.module.styl:4]: ?

**Pull request merged**